### PR TITLE
DM-26304: Relocate pexConfig formatter to here from daf_butler and test it

### PR DESCRIPTION
**Pull request by timj**

### CODE REVIEW

File: python/lsst/obs/base/formatters/pexConfig.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    extension = ".py"
+
+    def _readFile(self, path: str, pytype: Optional[Type[Any]] = None) -> Any:
+        """Read a pex.config.Config instance from the given file.
+
+        Parameters
+        ----------
+        path : `str`
+            Path to use to open the file.
+        pytype : `class`
```
Comment (TallJimbo) [inline python/lsst/obs/base/formatters/pexConfig.py:50]: `class` -> `type`?

File: python/lsst/obs/base/formatters/pexConfig.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+            Class to use to read the config file.
+
+        Returns
+        -------
+        data : `lsst.pex.config.Config`
+            Instance of class `pytype` read from config file. None
```
Comment (TallJimbo) [inline python/lsst/obs/base/formatters/pexConfig.py:56]: `pytype` should have double backticks.  `None` should have single backticks.

File: python/lsst/obs/base/formatters/pexConfig.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        """
+        if pytype is None:
+            raise RuntimeError("A python type is always required for reading pex_config Config files")
+
+        if not os.path.exists(path):
+            return None
+        instance = pytype()
+        # Configs can only be loaded if you use the correct derived type,
+        # but we can't really store the correct derive type in the StorageClass
+        # because that"d be a huge proliferation of StorageClasses.
```
Comment (TallJimbo) [inline python/lsst/obs/base/formatters/pexConfig.py:67]: `"` -> `'`

File: python/lsst/obs/base/formatters/pexConfig.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        try:
+            instance.load(path)
+            return instance
+        except AssertionError as err:
+            msg = str(err)
+            if not msg.startswith("config is of type"):
+                raise RuntimeError("Unexpected assertion; cannot infer Config class type.") from err
+            actualPyTypeStr = msg.split()[-1]
+            import sys
+            print("HHHEEELLO:::::::::::::::", msg, file=sys.stderr)
```
Comment (TallJimbo) [inline python/lsst/obs/base/formatters/pexConfig.py:81]: Um, yikes.

**Comment by timj** [comment]
Yes, the docstring issues were all there before. I have fixed many of them on DM-26008.

**Pull request merged**